src/archetype/createProject/index.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { SelectArchetypeStep } from "./SelectArchetypeStep";
+import { SpecifyArchetypeVersionStep } from "./SpecifyArchetypeVersionStep";
+import { SpecifyArtifactIdStep } from "./SpecifyArtifactIdStep";
+import { SpecifyGroupIdStep } from "./SpecifyGroupIdStep";
+import { SpecifyTargetFolderStep } from "./SpecifyTargetFolderStep";
+import { IProjectCreationMetadata, IProjectCreationStep, StepResult } from "./types";
+
+export const selectArchetypeStep = new SelectArchetypeStep();
+export const specifyArchetypeVersionStep = new SpecifyArchetypeVersionStep();
+export const specifyGroupIdStep = new SpecifyGroupIdStep();
+export const specifyArtifactIdStep = new SpecifyArtifactIdStep();
+export const specifyTargetFolderStep = new SpecifyTargetFolderStep();
+
+/**
+ * Run a group of steps to create a Maven project.
+ * @param steps array of IProjectCreationStep to run
+ * @param metadata stores data across steps
+ * @returns whether all steps are successfully passed
+ */
+export const runSteps = async (steps: IProjectCreationStep[], metadata: IProjectCreationMetadata): Promise<boolean> => {
+  if (!(steps[0] instanceof SelectArchetypeStep)) {
+    throw new Error("First step must be SelectArchetypeStep.");
+  }
+
+  for (let i = 0; i < steps.length; i += 1) {
+    steps[i].nextStep = steps[i + 1];
+    steps[i].previousStep = steps[i - 1];
```

Comment:
Will index of of bound here?

Reply:
Yes, but doesn't matter.  Suppose n = steps.length, then steps[-1] = steps[n] = undefined.